eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MultiReaderFastList.java

```diff
@@ -793,6 +794,34 @@ public T get(int index)
     }
 
     @Override
+    public Optional<T> getFirstOptional()
+    {
+        this.acquireReadLock();
+        try
+        {
+            return this.getDelegate().getFirstOptional();
+        }
+        finally
+        {
+            this.unlockReadLock();
+        }
+    }
+
+    @Override
+    public Optional<T> getLastOptional()
+    {
+        this.acquireReadLock();
+        try
+        {
+            return this.getDelegate().getLastOptional();
+        }
+        finally
+        {
+            this.unlockReadLock();
+        }
+    }
+
+    @Override
```

Comment:
`AbstractMultiReaderMutableCollection` does not implement `OrderedIterable` (see above).